nullaway/src/main/java/com/uber/nullaway/handlers/Handler.java

```diff
@@ -342,6 +343,28 @@ MethodInvocationNode onCFGBuildPhase1AfterVisitMethodInvocation(
       MethodInvocationTree tree,
       MethodInvocationNode originalNode);
 
+  /**
+   * Called to determine when a method acts as a cast-to-non-null operation on its parameters.
+   *
+   * <p>See {@link LibraryModels#castToNonNullMethods()} for more information about general
+   * configuration of <code>castToNonNull</code> methods.
+   *
+   * @param analysis A reference to the running NullAway analysis.
+   * @param state The current visitor state.
+   * @param methodSymbol The method symbol for the potential castToNonNull method.
+   * @param actualParams The actual parameters from the invocation node
+   * @param castToNonNullPositions The list of parameters for the method for which the method acts
+   *     as a cast, as computed by upstream handlers (the core analysis supplies a default set to
+   *     the first handler in the chain, based on CLI config).
+   * @return Updated parameter list computed by this handler
+   */
+  ImmutableSet<Integer> castToNonNullArgumentPositionsForMethod(
```

Comment:
I feel like this interface is too general.  Let's assume that for any given method, the handlers will return at most one argument position that should be treated as being cast to `@NonNull`.  Could we then have this method return `@Nullable Integer` rather than `ImmutableSet<Integer>`?  We could check for contradictions in the composite handler and just fail there if we ever see one.

Reply:
Fair enough. This is based on what was easy to implement based on what library models and other handler extension points already do, but, you are right, the semantics of `castToNonNull` with multiple cast arguments is not very well defined... (I mean, for the purposes of this change it is clear what to do: "error if any of these arguments is known to be non-null", but then it isn't clear what the method should actually be returning).

For consistency with other handler APIs, we probably want `Optional<Integer>` here and then a check within the Library Models Handler that we never have multiple "castToNonNull" models for the same method signature.

Reply:
Now that we check our code with NullAway I'd be in favor of getting rid of the other `Optional`s 🙂 But I'm ok with using it here for consistency.

Reply:
Let's do `@Nullable Integer` for this one, then, and as a follow up PR I can switch the other one around too.